tests/test_repro.py

```diff
@@ -1404,3 +1405,40 @@ def test(self):
 
         expected_metrics_display = "{}: {}".format(metrics_file, metrics_value)
         self.assertIn(expected_metrics_display, self._caplog.text)
+
+
+class TestPersistentOutput(TestDvc):
+    def test_ignore_build_cache(self):
+        with open("immutable", "w") as fobj:
+            fobj.write("123456789")
+
+        warning = "Build cache is ignored when persisting outputs."
+
+        with self._caplog.at_level(logging.WARNING, logger="dvc"):
+            assert (
+                main(
```

Comment:
This is hard to read, just use an intermediate `ret = main()` and then `assert ret == 0`.